### PR TITLE
optimize mergeadd for sparse_adam,*test=kunlun

### PR DESCRIPTION
**Pull request by helen88**
### PR types

Function optimization

### PR changes

OPS

### Describe

优化sparse_adam mergeadd功能,提升性能

### CODE REVIEW

**Review by zhangyk0314 — DISMISSED** [review]

File: paddle/fluid/operators/math/selected_rows_functor.cc [hunk 1]
```diff
@@ -502,32 +502,32 @@ struct MergeAdd<platform::XPUDeviceContext, T> {
...
+    memory::Copy(context.GetPlace(), y_rows_data, platform::CPUPlace(),
+                 merge_rows.data(), ym * sizeof(int64_t));
+    memory::Copy(context.GetPlace(), x_rows_data, platform::CPUPlace(),
+                 input_rows.data(), xm * sizeof(int64_t));
+    int r =
+        xpu::merge_dup_rows<T, int64_t>(context.x_context(), x_data, y_data,
+                                        x_rows_data, y_rows_data, xm, n, ym);
+    PADDLE_ENFORCE_EQ(r, XPU_SUCCESS,
+                      platform::errors::External(
+                          "XPU merge_dup_rows API return wrong value[%d %s], ",
```
Comment (taixiurong) [inline paddle/fluid/operators/math/selected_rows_functor.cc]: 换用宏判断 PADDLE_ENFORCE_XDNN_SUCCESS

File: paddle/fluid/operators/math/selected_rows_functor.cc [hunk 2]
```diff
@@ -603,17 +595,26 @@ struct MergeAdd<platform::XPUDeviceContext, T> {
...
+                   merge_rows.data(), ym * sizeof(int64_t));
+      memory::Copy(context.GetPlace(), x_rows_data, platform::CPUPlace(),
+                   input_rows.data(), xm * sizeof(int64_t));
+      int r =
+          xpu::merge_dup_rows<T, int64_t>(context.x_context(), x_data, y_data,
+                                          x_rows_data, y_rows_data, xm, n, ym);
+      PADDLE_ENFORCE_EQ(
+          r, XPU_SUCCESS,
+          platform::errors::External(
+              "XPU merge_dup_rows API return wrong value[%d %s], ", r,
```
Comment (taixiurong) [inline paddle/fluid/operators/math/selected_rows_functor.cc]: 同上

File: paddle/fluid/operators/math/selected_rows_functor.cc (hunk 1 above)
Comment (helen88) [reply paddle/fluid/operators/math/selected_rows_functor.cc]: fixed

File: paddle/fluid/operators/math/selected_rows_functor.cc (hunk 2 above)
Comment (helen88) [reply paddle/fluid/operators/math/selected_rows_functor.cc]: fixed

**Review by zhangyk0314 — APPROVED** [review]

**Review by taixiurong — COMMENTED** [review]
LGTM

**Review by QingshuChen — APPROVED** [review]

**Pull request merged**